examples/HelloWorldExampleDS/HelloWorldServer.h

```diff
@@ -20,21 +20,41 @@
 #ifndef HELLOWORLDSERVER_H_
 #define HELLOWORLDSERVER_H_
 
-#include <fastrtps/fastrtps_fwd.h>
-#include <fastrtps/rtps/common/Locator.h>
+#include <fastdds/dds/domain/DomainParticipantListener.hpp>
+#include <fastdds/rtps/common/Locator.h>
+
+namespace eprosima
+{
+    namespace fastdds
+    {
+        namespace dds
+        {
+            class DomainParticipant;
+        }
+    }
+}
 
 class HelloWorldServer
 {
 public:
     HelloWorldServer();
     virtual ~HelloWorldServer();
     //!Initialize the subscriber
-    bool init(eprosima::fastrtps::rtps::Locator_t server_address);
+    bool init(eprosima::fastdds::rtps::Locator server_address);
     //!RUN the subscriber
     void run();
 
 private:
-    eprosima::fastrtps::Participant* mp_participant;
+    class PubListener :public eprosima::fastdds::dds::DomainParticipantListener
+    {
+    public:
+        PubListener() :n_matched(0), firstConnected(false) {};
+        ~PubListener() {};
+        void on_participant_discovery(eprosima::fastdds::dds::DomainParticipant* participant, eprosima::fastrtps::rtps::ParticipantDiscoveryInfo&& info);
+        int n_matched;
+        bool firstConnected;
+    } * m_listener;
```

Comment:
This listener does not seem to be used in the example. The callback is not even implemented. Please remove if it is not being used.